### ISSUE TITLE
Netplan and Ubuntu 18.04

### ISSUE THREAD

**Issue by praisebob** [hashicorp/terraform-provider-vsphere#526]
Hi,

Using Terraform to provision Ubuntu VM's using the vSphere provider. When creating Ubuntu 16.04LTS VM's it would correctly populate the /etc/network/interfaces file with correct values. When creating Ubuntu 18.04 VM's Terraform is still populating /etc/network/interfaces rather than /etc/netplan/50-cloud-init.yaml.

Does vsphere provider support setting up network via netplan at all?

### Terraform Version

Terraform v0.11.7

### vSphere Provider Version

provider.vsphere v1.5.0

### Affected Resource(s)

Please list the resources as a list, for example:
- `vsphere_virtual_machine`

If this issue appears to affect multiple resources, it may be an issue with
Terraform's core, so please mention this.

### Terraform Configuration Files

```hcl
# configure the vSphere provider
provider "vsphere" {
  version = "~> 1.5"
  vsphere_server = "${var.vsphere_server}"
  user = "${var.vsphere_user}"
  password = "${var.vsphere_password}"

  allow_unverified_ssl = true
}

# configure the vSphere dc
data "vsphere_datacenter" "dc" {
  name = "Cool-DC"
}

# configure the vSphere datastore
data "vsphere_datastore" "datastore" {
  name = "Cool-storage"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

# configure the vSphere pool
data "vsphere_resource_pool" "pool" {
  name = "Cluster1/Resources"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

# configure the vSphere network
data "vsphere_network" "public" {
  name = "VLAN2"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

# configure the vSphere template
data "vsphere_virtual_machine" "template" {
  name = "ubuntu-18.04-template-tmp2"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

# vm's
resource "vsphere_virtual_machine" "cool-name" {
  name = "cool-name.cool.domain"
  folder = "cool/directory"
  resource_pool_id = "${data.vsphere_resource_pool.pool.id}"
  datastore_id = "${data.vsphere_datastore.datastore.id}"

  num_cpus = 1
  memory = 1024
  memory_reservation = 1024
  guest_id = "ubuntu64Guest"

  scsi_type = "${data.vsphere_virtual_machine.template.scsi_type}"

  network_interface {
    network_id = "${data.vsphere_network.public.id}"
    adapter_type = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }

  disk {
    label            = "disk0"
    size             = "${data.vsphere_virtual_machine.template.disks.0.size}"
    eagerly_scrub    = "${data.vsphere_virtual_machine.template.disks.0.eagerly_scrub}"
    thin_provisioned = "${data.vsphere_virtual_machine.template.disks.0.thin_provisioned}"
  }

  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"

    customize {
      linux_options {
        host_name = "cool-name"
        domain = "cool.domain"
      }

      network_interface {
        ipv4_address = "10.10.10.20"
        ipv4_netmask = 24
      }

      ipv4_gateway = "10.10.10.254"
    }
  }
}
```

**Comment by spstratis** [comment]
@praisebob Did you ever come up with a solution for this? I'm running into the same issue

**Comment by praisebob** [comment]
No, the project is put on a backburner since.

**Comment by Vrodi** [comment]
Hello, I read than upgrading to vSphere 6.5 U2 or higher solves the problem. I will try this.

**Comment by spstratis** [comment]
@Vrodi Please let me know if it works out for you.

**Comment by rismoney** [comment]
not sure if this is related, but in order for me to get things working (specifically dhcp), i had to guarantee that the template I was deploying from had the same port configuration as the new system I was deploying.  in the case of ens160 in the template:

```
  extra_config = {
    "ethernet0.pciSlotNumber" = "160"
  }
```

Additionally I had to ensure machine-id was wiped out in 2 locations etc/machine-id and in lib/dbus.  I am using the latest version of openvmtools.  I also had to configure the netplan to use mac address for dhcp, since it was using some weird 36 character string.

I was testing all this on vs6.0
